HW2/P3/mandelbrot.pyx

```diff
@@ -31,41 +49,113 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
     assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
 
-    with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
-                for iter in range(max_iterations):
-                    if magnitude_squared(z) > 4:
-                        break
-                    z = z * z + c
-                out_counts[i, j] = iter
+    for i in prange(in_coords.shape[0], nogil=True, num_threads=threads, schedule=static):
+        for j in range(in_coords.shape[1]):
+            c = in_coords[i, j]
+            z = 0
+            for iter in range(max_iterations):
+                if magnitude_squared(z) > 4: #then his in AVX
+                    break
+                z = z * z + c #start with coding this in AVX
+            out_counts[i, j] = iter
+
+
+
 
 
 
 # An example using AVX instructions
-cpdef example_sqrt_8(np.float32_t[:] values):
+cpdef mandelbrot_AVX(np.complex64_t [:,:] in_coords, np.uint32_t [:,:] out_counts, int threads):
+
+
     cdef:
-        AVX.float8 avxval
+        AVX.float8 tmp_real, tmp_img, c, z, mask, xx, yy, real, img1
+        AVX.float8 img, mag, new_ones, xx_mul, yy_mul, xy_sub, z_mag_real
+        AVX.float8 z_mag_img, z_mag, z_real, z_img, xy_2_mul, two_xy_mul, xy_mul 
+        AVX.float8 counts
+        int i, j, it, k
+        int max_iterations=511
+        np.float32_t [:,:] coords_1 
+        np.float32_t [:,:] coords_2 
+        AVX.float8 zeros = AVX.float_to_float8(0)
+        AVX.float8 thresh = AVX.float_to_float8(4)
+        AVX.float8 two = AVX.float_to_float8(2)
+        AVX.float8 ones = AVX.float_to_float8(1)
         float out_vals[8]
         float [:] out_view = out_vals
+        
+
+    coords_1 = np.real(in_coords)
+    coords_2 = np.imag(in_coords)
+
+
+
+    for j in prange(in_coords.shape[0], nogil=True, num_threads=threads, schedule=static):
+      for i in range(0, in_coords.shape[1], 8):
+
+        #pull out the real components
+        tmp_real = AVX.make_float8(coords_1[j, i],coords_1[j, i+1],coords_1[j, i+2],coords_1[j, i+3],coords_1[j, i+4],coords_1[j, i+5],coords_1[j, i+6],coords_1[j, i+7])
+
+        #pull out the imaginary components
+        tmp_img = AVX.make_float8(coords_2[j, i],coords_2[j, i+1],coords_2[j, i+2],coords_2[j, i+3],coords_2[j, i+4],coords_2[j, i+5],coords_2[j, i+6],coords_2[j, i+7])
+
+
+        #initialize all of the vectors
+        z_real = zeros
+        z_img = zeros
+        z_mag_real = zeros
+        z_mag_img = zeros
+        z = zeros
+        counts = zeros
+        z_mag = AVX.add(z_mag_real, z_mag_img)
+
+
+        for it in range(max_iterations):
+
+          #create the mask which acts as the if statement
+          mask = AVX.less_than(z_mag, thresh) 
+
+          if not AVX.signs(mask): #this will ensure that the loop breaks if the conditions are no longer met
```

Comment:
You have a different stop condition than mine. I had to compare signs with 255. Maybe the problem is here?